app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdUpdate.kt

```diff
@@ -33,15 +34,23 @@ class ScanSimpleIdUpdate @Inject constructor(
       event: PatientSearchByIdentifierCompleted
   ): Next<ScanSimpleIdModel, ScanSimpleIdEffect> {
     val scanResult = if (event.patients.isNotEmpty()) {
-      val patientId = event.patients.first().uuid
-      PatientFound(patientId)
+      patientFoundByIdentifierSearch(patients = event.patients, identifier = event.identifier)
     } else {
       PatientNotFound(event.identifier)
     }
 
     return next(model = model.notSearching(), SendScannedIdentifierResult(scanResult))
   }
 
+  private fun patientFoundByIdentifierSearch(patients: List<Patient>, identifier: Identifier): ScanResult {
+    return if (patients.size > 1) {
+      EnteredShortCode(BpPassport.shortCode(identifier))
+    } else {
+      val patientId = patients.first().uuid
+      PatientFound(patientId)
+    }
```

Comment:
It took me a while to realise that the only reason we are doing this is because the `ShortCodeSearchResultScreen` expects a short code as input.

This works for now, but I would recommend a better way to do it would be to change `ShortCodeSearchResultScreen` to accept multiple types of input (short codes/full identifiers).

Reply:
That way, it can also handle types of identifiers that may not have the notion of a short code (like the upcoming NHID).

Reply:
Yeah, good point. Will look into the improvements we can make.

Reply:
One of the reasons why I was fine with leaving it as the string is, we can just pass the identifier value and compare it across the different identifiers to show the results.